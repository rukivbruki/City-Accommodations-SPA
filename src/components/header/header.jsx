import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {Link} from "react-router-dom";
import {getAuthorizationStatus, getUserData} from "../../reducer/user/selector";
import {PageAddress, REQUEST} from "../../constants";

const Header = ({isAuthorizationRequired, userData}) => {
  return (
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <div className="header__left">
            <Link to="/" className="header__logo-link header__logo-link--active">
              <img className="header__logo" src="/img/logo.svg" alt="6 cities logo" width="81" height="41"/>
            </Link>
          </div>
          <nav className="header__nav">
            <ul className="header__nav-list">
              <li className="header__nav-item user">
                <Link
                  to={isAuthorizationRequired ? PageAddress.LOGIN : PageAddress.FAVORITE}
                  className="header__nav-link header__nav-link--profile"
                >
                  <div
                    className="header__avatar-wrapper user__avatar-wrapper"
                    style={isAuthorizationRequired ? {} : {backgroundImage: `url(${REQUEST.BASE_URL}${userData.avatar})`}}
                  >
                  </div>
                  <span className="header__user-name user__name">
                    {isAuthorizationRequired ? `Sign In` : userData.email}
                  </span>
                </Link>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
  );
};

Header.propTypes = {
  isAuthorizationRequired: PropTypes.bool.isRequired,
  userData: PropTypes.shape({
    avatar: PropTypes.string,
    email: PropTypes.string,
    id: PropTypes.number,
    isPro: PropTypes.bool,
    name: PropTypes.string,
  }),
};


const mapStateToProps = (state) => ({
  isAuthorizationRequired: getAuthorizationStatus(state),
  userData: getUserData(state)
});


export {Header};
export default connect(mapStateToProps)(Header);
