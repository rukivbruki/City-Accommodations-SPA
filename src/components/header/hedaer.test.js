import React from "react";
import {shallow, configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {Header} from "./header";

configure({adapter: new Adapter()});

describe(`Header component should render is correctly`, () => {
  it(`if user require authorization`, () => {
    const header = shallow(<Header
      isAuthorizationRequired={true}
      userData={{}}
    />);

    expect(header).toMatchSnapshot();
  });

  it(`if user already authorization`, () => {
    const header = shallow(<Header
      isAuthorizationRequired={false}
      userData={{
        avatar: `static/avatar/10.jpg`,
        email: `Shramko.web@yahoo.com`,
        id: 13,
        isPro: false,
        name: `Shramko.web`,
      }}
    />);

    expect(header).toMatchSnapshot();
  });
});
