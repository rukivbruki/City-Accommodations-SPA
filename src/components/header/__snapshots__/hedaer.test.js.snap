// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render is correctly if user already authorization 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <Link
          className="header__logo-link header__logo-link--active"
          to="/"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="/img/logo.svg"
            width="81"
          />
        </Link>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <Link
              className="header__nav-link header__nav-link--profile"
              to="/favorite"
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
                style={
                  Object {
                    "backgroundImage": "url(https://htmlacademy-react-2.appspot.com/six-citiesstatic/avatar/10.jpg)",
                  }
                }
              />
              <span
                className="header__user-name user__name"
              >
                Shramko.web@yahoo.com
              </span>
            </Link>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Header component should render is correctly if user require authorization 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <Link
          className="header__logo-link header__logo-link--active"
          to="/"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="/img/logo.svg"
            width="81"
          />
        </Link>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item user"
          >
            <Link
              className="header__nav-link header__nav-link--profile"
              to="/login"
            >
              <div
                className="header__avatar-wrapper user__avatar-wrapper"
                style={Object {}}
              />
              <span
                className="header__user-name user__name"
              >
                Sign In
              </span>
            </Link>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
